### PR TITLE
Add GitHub branch protection helpers, get/update/delete.

### PR DESCRIPTION
**Pull request by mikenairn**
* Add separate GitHub Utils class.

**ghGetBranchProtection**

Get a branches protection status:

```
ghGetBranchProtection('master', 'feedhenry', 'fh-mbaas')
```
**ghUpdateBranchProtection**

Update a branches protection settings:
```
ghUpdateBranchProtection('master', 'feedhenry', 'fh-mbaas', '{"required_status_checks": {"strict": true, "contexts": ["continuous-integration/jenkins/pr-head"]},"enforce_admins": true, "restrictions": null}')
```
**ghRemoveBranchProtection**

Remove a branches protection settings:
```
ghUpdateBranchProtection('master', 'feedhenry', 'fh-mbaas')
```

### CODE REVIEW

File: src/org/feedhenry/Utils.groovy [hunk 1]
```diff
@@ -45,3 +45,15 @@ static def getDate() {
     SimpleDateFormat yearMonthDateHourMin = new SimpleDateFormat("yyyyMMddHHmm")
     return yearMonthDateHourMin.format(now)
 }
+
+String doCurlCmd(url, httpMethod = 'GET', postData = null, credentialsId = 'githubautomatron') {
+    String ghStatusRaw = ''
```
Comment (grdryn) [inline src/org/feedhenry/Utils.groovy]: Consider removing the `gh` prefix

**Review by grdryn — APPROVED** [review]
It's unlikely, but maybe possible, that the node that such a step would run on, may not have curl. As a result, I found this plugin: https://jenkins.io/doc/pipeline/steps/http_request/

I'm not sure if it could do what you want as easily as curl, or at all...just pointing it out. It might not be worth trying to use it even.

File: src/org/feedhenry/Utils.groovy (hunk 1 above)
Comment (mikenairn) [reply src/org/feedhenry/Utils.groovy]: Good catch, missed during my refactor.

**Comment by mikenairn** [comment]
@grdryn I changed this to use the http_request plugin https://github.com/feedhenry/fh-pipeline-library/pull/34/commits/2bd54c8afe2d07458cf5d2891eb6640a7de50445

File: src/org/feedhenry/GitHubUtils.groovy [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/groovy
+package org.feedhenry
+
+import groovy.json.JsonSlurperClassic
+
+def ghApiRequest(String endpoint, String httpMethod = 'GET', String requestBody = null, String credentialsId = 'githubautomatron', customHeaders = []) {
```
Comment (grdryn) [inline src/org/feedhenry/GitHubUtils.groovy:6]: Not a big deal, but do we know of a suitable return type for this? I just noticed that the method after this has a return type of String, but this is just a def.

**Review by grdryn — APPROVED** [review]

**Pull request merged**